netket/nn/symmetric_linear.py

```diff
@@ -101,62 +101,446 @@ def full_bias(self, bias):
+    """Initializer for the bias."""
 
-    The G-convolution generalizes the convolution to non-commuting groups:
+    def setup(self):
+        self.n_cells = np.product(np.asarray(self.shape))
+        self.n_symm = len(np.asarray(self.space_group)) // self.n_cells
+        self.sites_per_cell = np.asarray(self.space_group).shape[1] // self.n_cells
 
-    .. math ::
+        self.stdev = 1.0 / np.sqrt(self.n_cells * self.sites_per_cell)
 
-        C^i_g = \sum_h {\bf W}_{g^{-1} h} \cdot {\bf f}_h
+        self.mapping = (
+            np.asarray(self.space_group)[:, : self.sites_per_cell]
+            .reshape(self.n_cells, self.n_symm, self.sites_per_cell)
+            .transpose(1, 2, 0)
+            .reshape(self.n_symm, self.sites_per_cell, *self.shape)
+        )
 
-    Group elements that differ by the same symmetry operation (i.e. :math:`g = xh`
-    and :math:`g' = xh'`) are connected by the same filter.
+    def make_kernel(self, kernel):
+        kernel = kernel[..., self.mapping]
+
+        return kernel
+
+    @compact
+    def __call__(self, x: Array) -> Array:
+        """Applies the equivariant transform to the inputs along the last two
+        dimensions (-2: features, -1: group elements)
+        """
+
+        dtype = jnp.promote_types(x.dtype, self.dtype)
+        x = jnp.asarray(x, dtype)
+
+        x = (
+            x.reshape(-1, self.n_cells, self.sites_per_cell)
+            .transpose(0, 2, 1)
+            .reshape(-1, self.sites_per_cell, *self.shape)
+        )
+
+        if self.kernel_init:
+            kernel = self.param(
+                "kernel",
+                self.kernel_init,
+                (self.features, self.n_cells * self.sites_per_cell),
+                self.dtype,
+            )
+        else:
+            kernel = self.param(
+                "kernel",
+                normal(self.stdev),
+                (self.features, self.n_cells * self.sites_per_cell),
+                self.dtype,
+            )
+
+        kernel = jnp.asarray(kernel, dtype)
+
+        if self.mask:
+            kernel = kernel * jnp.expand_dims(self.mask, 0)
+
+        kernel = self.make_kernel(kernel)
+
+        x = jnp.fft.fftn(x, s=self.shape).reshape(*x.shape[:2], self.n_cells)
+
+        kernel = jnp.fft.fftn(kernel, s=self.shape).reshape(
+            *kernel.shape[:3], self.n_cells
+        )
+
+        x = lax.dot_general(
+            x, kernel, (((1,), (2,)), ((2,), (3,))), precision=self.precision
+        )
+        x = x.transpose(1, 2, 3, 0)
+        x = x.reshape(*x.shape[:3], *self.shape)
+
+        x = jnp.fft.ifftn(x, s=self.shape).reshape(*x.shape[:3], self.n_cells)
+        x = x.transpose(0, 1, 3, 2).reshape(*x.shape[:2], -1)
+
+        if self.use_bias:
+            if self.bias_init:
+                bias = self.param("bias", self.bias_init, (self.features, 1), dtype)
+            else:
+                bias = self.param("bias", zeros, (self.features, 1), dtype)
```

Comment:
```suggestion
        if self.use_bias:
            if self.bias_init:
                bias = self.param("bias", self.bias_init, (self.features, 1), self.dtype)
            else:
                bias = self.param("bias", zeros, (self.features, 1), self.dtype)
```
You don't want this bit to depend on the input!